pkg/kubelet/kubelet.go

```diff
@@ -2845,6 +2854,110 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}, 100*time.Millisecond, stopCh)
 }
 
+// handleMaxContainerCheckpoints goes through the list of checkpoints of a
+// container in a pod and deletes old checkpoints if there are more than
+// 'kl.maxContainerCheckpoints'.
+func (kl *Kubelet) handleMaxContainerCheckpoints(
+	podUID types.UID,
+	podFullName,
+	containerName string,
+	options *runtimeapi.CheckpointContainerRequest,
+) error {
+
+	if kl.maxContainerCheckpoints < 1 {
+		// This cannot really happen as the kubelet config parser will
+		// error out early if this is < 1. This additional error
+		// handling is mainly here to make sure tests are working.
+		return errors.New("maxContainerCheckpoints needs to be >= 1")
+	}
+
+	globPattern := filepath.Join(
+		kl.getCheckpointsDir(),
+		fmt.Sprintf(
+			// This glob pattern tries to match RFC3339 (including leaf seconds)
+			"checkpoint-%s-%s-[0-9][0-9][0-9][0-9]-[0-2][0-9]-[0-3][0-9]T[0-2][0-9]:[0-5][0-9]:[0-6][0-9]*.tar",
+			podFullName,
+			containerName,
+		),
+	)
+
+	// Lock while accessing the checkpoints on disk to avoid chaos
+	// if multiple threads would do it at the same time.
+	kl.updateMaxContainerCheckpoints.Lock()
+	defer kl.updateMaxContainerCheckpoints.Unlock()
+
+	checkpointArchives, err := filepath.Glob(globPattern)
+
+	if err != nil {
+		return fmt.Errorf(
+			"error looking for checkpoint archives with pattern %s: %w",
+			globPattern,
+			err,
+		)
+	}
+
+	checkpointArchivesCounter := len(checkpointArchives)
+
+	if checkpointArchivesCounter <= int(kl.maxContainerCheckpoints) {
```

Comment:
What's the point of having maxContainerCheckpoints of different type?

Reply:
I was trying to keep the same type as defined in `types.go`, which seems to be limited to a few certain types, as long as possible. `len()` returns something else and at some point I have to cast to use the result of `len()` to compare it to `int32`.

Reply:
Makes sense. Thank you for the explanations.